drivers/modem/gsm_ppp.c

```diff
@@ -581,30 +584,32 @@ static void rssi_handler(struct k_work *work)
 
 }
 
-static void gsm_finalize_connection(struct gsm_modem *gsm)
+static void gsm_finalize_connection(struct k_work *work)
 {
 	int ret = 0;
+	struct k_work_delayable *dwork = k_work_delayable_from_work(work);
+	struct gsm_modem *gsm = CONTAINER_OF(dwork, struct gsm_modem,
+					     gsm_configure_work);
 
 	/* If already attached, jump right to RSSI readout */
-	if (gsm->attached) {
```

Comment:
I might split these into different modem states and do it in different work item so that it is cleanner and doesn't have to jump around at the top, what do you all think?

@jukkar @bbilas @tautologyclub @mniestroj @endian-benjamin @rlubos

Reply:
My preference would be to get this merged. Make a new PR with further improvements